front/components/workspace/ProviderManagementModal.tsx

```diff
@@ -65,19 +67,31 @@ export function ProviderManagementModal({
     },
     [setProviderStates]
   );
+
+  const handleSave = useCallback(() => {
+    const activeProviders = MODEL_PROVIDER_IDS.filter(
+      (key) => providerStates[key]
+    );
```

Comment:
Not sure if the following is actually preferred ?
```typescript
const providerStatesKeys = Object.keys(providerStates) as ModelProviderIdType[];
const activeProviders = providerStatesKeys.filter(
   (key) => providerStates[key]
);
```

Reply:
Use of as is to be avoided as much as possible 👍